runtime/serde/serde-json/common/test/aws/smithy/kotlin/runtime/serde/json/JsonStreamWriterTest.kt

```diff
@@ -9,68 +9,231 @@ import kotlin.test.assertEquals
 
 @OptIn(ExperimentalStdlibApi::class)
 class JsonStreamWriterTest {
-
     @Test
-    fun checkJsonSerializesCorrectlyWithWrapper() {
-        val msg1 = Message(
-            912345678901,
-            "How do I stream JSON in Java?",
-            null,
-            user1
-        )
-        val msg2 = Message(
-            912345678902,
-            "@json_newb just use JsonWriter!",
-            arrayOf(50.454722, -104.606667),
-            user2
-        )
-        assertEquals(
-            expected,
-            writeJsonStream(
-                listOf(msg1, msg2)
-            )?.decodeToString()
-        )
+    fun testArrayOfObjects() {
+        // language=JSON
+        val expected = """[
+    {
+        "id": 912345678901,
+        "text": "How do I stream JSON in Java?",
+        "geo": null,
+        "user": {
+            "name": "json_newb",
+            "followers_count": 41
+        }
+    },
+    {
+        "id": 912345678902,
+        "text": "@json_newb just use JsonWriter!",
+        "geo": [
+            50.454722,
+            -104.606667
+        ],
+        "user": {
+            "name": "jesse",
+            "followers_count": 2
+        }
     }
```

Comment:
**Nit**: We should probably write our own test data instead of reusing examples from https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.0/com/google/gson/stream/JsonWriter.html

Reply:
Ha didn't even know that is where it was from. This test was originally written by Zeynep who was briefly on the team. I'll remove and write our own